packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -509,10 +503,17 @@ export class ConnectionService {
     connectionRecord.state = newState
     await this.connectionRepository.update(connectionRecord)
 
+    this.emitStateChangedEvent(connectionRecord, previousState)
+  }
+
+  private emitStateChangedEvent(connectionRecord: ConnectionRecord, previousState: DidExchangeState | null) {
```

Comment:
Why not 
```suggestion
  private emitStateChangedEvent(connectionRecord: ConnectionRecord, previousState?: DidExchangeState) {
```

Reply:
Events are emitted with previousState value null (which has a different meaning than undefined). If we change this now it would be a breaking change to events (because `previousState` will now be undefined)